store/modules/events.ts

```diff
@@ -0,0 +1,66 @@
+type CommunityEvent = {
+  type: String,
+  title: String,
+  image: String,
+  place: String,
+  location: String,
+  date: String,
+  to: String
+}
+
+export default {
+  state: {
+    communityEvents: [],
+    typeFilters: [],
+    locationFilters: []
+  },
+  getters: {
+    typeFilters (state) {
+      return state.typeFilters
+    },
+    locationFilters (state) {
+      return state.locationFilters
+    },
+    filteredEvents (state) {
+      const { communityEvents, typeFilters, locationFilters } = state
+      const events = communityEvents
+      const noTypeFilters = typeFilters.length === 0
+      const noLocationFilters = locationFilters.length === 0
+
+      if (noTypeFilters && noLocationFilters) { return events }
+
+      const eventsAfterApplyTypeFilter = filterBy(events, typeFilters, 'type')
+
+      return filterBy(eventsAfterApplyTypeFilter, locationFilters, 'location')
+
+      function filterBy (allEvents, selectedFilters, propToFilter) {
+        const noFilters = selectedFilters.length === 0
+
+        if (noFilters) { return allEvents }
+
+        return allEvents.filter(event => selectedFilters.includes(event[propToFilter]))
+      }
+    }
+  },
+  mutations: {
+    setEvents (state, events) {
+      state.communityEvents = events
+    },
+    addFilter (state, payload) {
+      const { filter, filterValue } = payload
+
+      state[filter].push(filterValue)
+    },
+    removeFilter (state, payload) {
+      const { filter, filterValue } = payload
+
+      state[filter].splice(state[filter].indexOf(filterValue), 1)
```

Comment:
Be careful, `.splice(start, 1)`  with a negative value for `start` will remove 1 item [starting from the end](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) and so, trying to remove a non-present item will remove the last filter.